compiler/rustc_middle/src/middle/privacy.rs

```diff
@@ -181,12 +181,15 @@ impl EffectiveVisibilities {
             if !matches!(tcx.def_kind(def_id), DefKind::Impl { .. }) {
```

Comment:
This exception should be extended to associated items in trait impls, because nominal visibility doesn't make sense for them as well.

Reply:
Fixed!

Reply:
Right now the ICE will still occur for associated constants and types, and we should keep the check for associated items in inherent impls, because there nominal visibilities make sense.

Reply:
Fixed!